docs/DeveloperGuide.md

```diff
@@ -650,42 +730,113 @@ Use case ends.
    Use case ends.
 
 * 3a. The given index is invalid.
-
   * 3a1. ATAS shows an error message.
+  
+   Use case resumes at step 2.
+
+* 4a. User denies confirmation
+
+   Use case ends.
+   
+
+</div>
 
-      Use case resumes at step 2.
+<div markdown="block" class="alert alert-info">
 
-**Use case: UC03 - Clear the class list**
+**Use case: UC03 - Edit a student**
+
+**Guarantees :**
+* Student will be edited only if the edited name, Matriculation number, NUS email and tags are valid.
+* The student list will not have duplicate students after edit.
 
 **MSS**
 
-1.  User requests to clear the class list
-1.  ATAS clears the class list
+1.  User requests to edit a student with name, Matriculation number, NUS email and tags with the student index of that student.
+1.  ATAS checks if the input is valid and scans through the student list for duplication.
+1.  ATAS sends confirmation message.
+1.  User confirms.
+1.  ATAS edits the student.
 
-Use case ends.
+   Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 2a. A student in the list has the same Matriculation number.
+   * 2b1. ATAS shows an error message.
+   
+   Use case ends.
+
+* 2b. Matriculation number provided is invalid.
+  * 2b1. ATAS shows an error message.
+  
+  Use case ends.
+  
+* 2c. A student in the list has the same NUS email.
+  * 2c1. ATAS shows an error message.
+  
+  Use case ends.
+  
+* 2d. The email provided is not NUS email.
+  * 2d1. ATAS shows an error message.
+  
+  Use case ends.
+  
+* 2e. The given student index is invalid.
+  * 2e1. ATAS shows an error message.
+  
+  Use case ends.
+  
+* 3a. User denies confirmation.
+
+  Use case ends.
+  
+</div>
+
+<div markdown="block" class="alert alert-info">
+
+**Use case: UC04 - Clear the class list**
```

Comment:
Maybe this could be renamed to student list to more accurately reflect the naming convention used?